docs/src/examples/sinusoid_example.md

```diff
@@ -0,0 +1,425 @@
+```julia
+using CalibrateEmulateSample.MarkovChainMonteCarlo
+```
+
+We will provide the API with the observations, priors and our cheap emulator from the previous section. In this 
+example we use the GP emulator. 
+
+```julia
+mcmc = MCMCWrapper(RWMHSampling(), y_obs, prior, emulator; init_params = init_sample)
+# First let's run a short chain to determine a good step size
+new_step = optimize_stepsize(mcmc; init_stepsize = 0.1, N = 2000, discard_initial = 0)
+
+# Now begin the actual MCMC
+println("Begin MCMC - with step size ", new_step)
+chain = MarkovChainMonteCarlo.sample(mcmc, 100_000; stepsize = new_step, discard_initial = 2_000)
+
+# We can print summary statistics of the MCMC chain
+display(chain)
+```
+
+| parameters         | mean    | std |
+| :---------------- | :------: | :----: |
+| amplitude         |   1.1049 | 0.0942  |
+| vert_shift        |   6.3753 | 0.4591  |
+
+Note that these values are provided in the unconstrained space. The vertical shift
+seems reasonable, but the amplitude is not. This is because the amplitude is constrained to be
+positive, but the MCMC is run in the unconstrained space.  We can transform to the real 
+constrained space and re-calculate these values.
+
+```julia
+# Extract posterior samples
+posterior = MarkovChainMonteCarlo.get_posterior(mcmc, chain)
+# Back to constrained coordinates
+constrained_posterior = Emulators.transform_unconstrained_to_constrained(
+    prior, MarkovChainMonteCarlo.get_distribution(posterior)
+)
+```
+This gives:
+
+
+| parameters         | mean    | std |
+| :---------------- | :------: | :----: |
+| amplitude         |   3.0324 | 0.2863  |
+| vert_shift        |   6.3753 | 0.4591  |
+
+This is in agreement with the true $\theta=(3.0, 7.0)$ and with the observational covariance matrix we provided $\Gamma=0.2 * I$ (i.e., a standard deviation of approx. $0.45$). `CalibrateEmulateSample.jl` has built-in plotting
+recipes to help us visualize the prior and posterior distributions.
+
+```julia
+# We can quickly plot priors and posterior using built-in capabilities
+p = plot(prior)
+plot!(posterior)
+
+```
+![GP_posterior](../assets/sinusoid_posterior_GP.png)
+
+
+The MCMC has learned the posterior distribution which is much narrower than the prior. 
+For multi-dimensional problems, we may be interested in how the posterior distributions for
+each parameter co-vary. We plot a 2D histogram of $\theta_1$ vs $\theta_2$ below. 
+
+![GP_2d_posterior](../assets/sinusoid_MCMC_hist_GP.png)
+
+
+### Sample with Random Features
+
+We can repeat the sampling method using the random features emulator instead of the Gaussian
+process and we find similar results: 
+
+![RF_2d_posterior](../assets/sinusoid_MCMC_hist_RF.png)
+
+It is reassuring to see that our uncertainty quantification methods are robust to the different emulator
+choices here. The GP and RF emulators showed similar accuracy during validation, and therefore the MCMC using 
```

Comment:
I think we should be honest here, and state that one can obtain poor results if the GP kernel or RF kernel structure is chosen to be too inflexible. So people should try several configurations out too